ext/json/ext/parser/parser.rl

```diff
@@ -573,7 +577,17 @@ static char *JSON_parse_string(JSON_Parser *json, char *p, char *pe, VALUE *resu
     if (json->symbolize_names && json->parsing_name) {
       *result = rb_str_intern(*result);
     } else if (RB_TYPE_P(*result, T_STRING)) {
+# if STR_UMINUS_DEDUPE_FROZEN
```

Comment:
@casperisfine Can you add the reason like https://github.com/msgpack/msgpack-ruby/pull/194/files#diff-4bc109d62ab9703449bc38b7290e3cacR164 ?

Reply:
Sure. However I recently discovered that this code isn't quite correct. I'll push a new version very soon https://github.com/ruby/spec/pull/788

Reply:
Done.